pkg/types/conversion/tfjson/tfjson.go

```diff
@@ -0,0 +1,221 @@
+/*
+ Copyright 2022 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package tfjson
+
+import (
+	tfjson "github.com/hashicorp/terraform-json"
+	schemav2 "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// GetV2ResourceMap converts input resource schemas with
+// "terraform-json" representation to terraform-plugin-sdk representation which
+// is what Terrajet expects today.
+//
+// What we are trying to achieve here is to convert a lower level
+// representation of resource schema map, e.g. output of `terraform providers schema -json`
+// to plugin sdk representation. This is mostly the opposite of what the
+// following method is doing: https://github.com/hashicorp/terraform-plugin-sdk/blob/7e0a333644f1971a936995677b7a106140a0659f/helper/schema/core_schema.go#L43
+//
+// Ideally, we should not rely on plugin SDK types in Terrajet at all but only
+// work with types in https://github.com/hashicorp/terraform-json which is
+// there exactly for this purpose, an external representation of Terraform
+// schemas. This conversion aims to be an intermediate step for that ultimate
+// goal.
+func GetV2ResourceMap(resourceSchemas map[string]*tfjson.Schema) map[string]*schemav2.Resource {
```

Comment:
I wonder how we can help provider builders to know that the old way is deprecated and that's how we'll do this going forward. `tjconfig.NewProvider` still accepts a resource map but the end goal for this work is to make it accept the CLI schema, maybe even as a `string`. Something like `tjconfig.NewProviderWithSchema` and deprecate `tjconfig.NewProvider`?

Reply:
With `tjconfig.NewProviderWithSchema(schema string, ...` calling `GetV2ResourceMap` and the rest would be same as `tjconfig.NewProvider`.

Reply:
That's a great point @muvaf!
Done!